src/DocRaptor.php

```diff
@@ -39,15 +40,17 @@ class ApiWrapper
     protected $baseurl;
 
 
-
     /**
+     * @param HttpTransferInterface $httpClient
      * @param string|null $api_key
      */
-    public function __construct($api_key = null)
+    public function __construct(HttpTransferInterface $httpClient, $api_key = null)
```

Comment:
My take on DI is never to constructor inject anything that is not an object without making it optional, actually I would prefer not to inject single parameters at all. If you work with automatic dependency tree scanning having the rule that `new` statements are only used at the very top of the dependency tree (bootstrap) then having single params injected into constructors makes your life really hard. So, if anything, I would make it possible to inject a `Config` object containing more than just the api key which is extracted in constructor but also optional.